CMakeLists.txt

```diff
@@ -47,6 +47,8 @@ option(ZIG_TEST_COVERAGE "Build Zig with test coverage instrumentation" OFF)
 # git log -p -- deps/lld
 option(ZIG_FORCE_EXTERNAL_LLD "If your system has the LLD patches use it instead of the embedded LLD" OFF)
 
+set(LLVM_STATIC off CACHE BOOL "Link LLVM statically")
```

Comment:
Let's put this next to the corresponding `ZIG_STATIC` option, rename it to `ZIG_STATIC_LLVM`, and improve the documentation string. "Prefer linking against static LLVM libraries".